scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt

```diff
@@ -50,6 +50,19 @@ import org.ossreviewtoolkit.model.config.ScannerConfiguration
  *   variables prefixed by "$" e.g. "$Var1_$Var2". Variables are also passed as options and are prefixed by
  *   [NAMING_CONVENTION_VARIABLE_PREFIX] e.g. namingVariableVar1 = "foo".
  * * **"namingScanPattern":** A pattern for scan names when scans are created on the FossID instance.
+ *
+ * URL mapping options. These options allow transforming the URLs of specific repositories before they are passed to
+ * the FossID service. This may be necessary if FossID uses a different mechanism to clone a repository, e.g. via SSH
```

Comment:
I'm not sure I can follow here: Isn't the mechanism to clone a repository implied by the URL / protocol? So if FossID is passed a HTTP URL, how can it still decide by itself to use a different mechanism like SSH?

Reply:
This is a ,misunderstanding: FossID tries to clone exactly the URL it is passed. However, in our setup ORT and FossID run in different network zones, and in some cases the URL used by ORT to clone a repository does not work for FossID. Therefore, with this new mechanism the URL is transformed accordingly by ORT before it is propagated to FossID.

Reply:
As the idea / wish to rewrite download / clone URLs has occurred several times in ORT's history, IMO we should think about implementing this in a higher level, so that URLs are always being rewritten, independent of the scanner being used.

Would that also work for you, or would such a global rewrite break something else for your network setup?

Reply:
Hm, in our case there is indeed a scanner-specific aspect. FossID uses a different URL than the ORT downloader invoked from the ScanCode abstraction.

Reply:
@sschuberth, how to proceed here? Could we use this implementation as a kind of starting point to be improved later? Or do you have something different in mind?

Reply:
> Or do you have something different in mind?

What I originally had in mind was to implement this in the analyzer, and perform some VCS URL mapping in the process of creating `vcsProcessed` from `vcs`. However, as I understand your use-case that would not work, as for some scanners you need the original URL, and for others the mapped URL, correct?

So the mapping basically depends on the infrastructure the scanner runs on. Which means it probably doesn't even make sense to move the mapping logic to a more common place like `ProvenanceScannerWrapper`.

Reply:
I'm afraid this is complicating the current (and future added) logic for handling provenance / resolved provenance.
I believe a complication could be avoided if this remained completely transparent: e.g. would it make sense that the scan result returned from the scanner would still return the original repository URL (not the mapped one) so that the generic code can remain completely unaware ?

Reply:
> would it make sense that the scan result returned from the scanner would still return the original repository URL (not the mapped one) so that the generic code can remain completely unaware ?

I makes me feel a bit uneasy that strictly speaking we're then not returning the URL / provenance that has been scanned in this case... which is esp. bad if you have some error in your defined mapping, and scan some completely different than you intend to.

Reply:
With the current implementation in this PR, the URL is only transformed before it is passed to the FossID server. The URL reported in the provenance of the scan result is still the original one. I am not sure, which one to prefer. At least in our case, the same repository should be accessed, only with a different protocol.